src/NeoSharp.Core/ExtensionMethods/MessageExtensionMethods.cs

```diff
@@ -1,21 +1,17 @@
-﻿using System;
-using System.Linq;
-using NeoSharp.Core.Messaging;
-using NeoSharp.Core.Messaging.Messages;
+﻿using NeoSharp.Core.Messaging;
 
 namespace NeoSharp.Core.ExtensionMethods
 {
     public static class MessageExtensionMethods
     {
-        private static readonly Type[] _handshakeMessageTypes =
-        {
-            typeof(VersionMessage),
-            typeof(VerAckMessage)
-        };
-
+        /// <summary>
+        /// Is this message a handshake message?
+        /// </summary>
+        /// <param name="message">Message</param>
+        /// <returns>Return true or false</returns>
         public static bool IsHandshakeMessage(this Message message)
         {
-            return _handshakeMessageTypes.Contains(message.GetType());
+            return message.Command == MessageCommand.verack || message.Command == MessageCommand.version;
```

Comment:
Our contract for handshake message is not only right command field. I think old version expressed better the intention.

Reply:
What other messages should be a handshake message?

Reply:
It is only 2 messages for now VersionMessage and VerAckMessage. What I meant is if you pass message with message.Command == version, for example, it would not be sufficient to expect message contains VersionPayload. But if you check the type in this case you can be sure yes it is properly constructed handshake message.

Reply:
Is faster and at low level the real message is the command not the type

Reply:
I knew you would play this card :wink: Just remember we rewrite NEO project because of readability and testability not because of the speed. Please favor more Domain-Driven Design then operations on primitives.